src/main/java/dev/unnm3d/rediseconomy/currency/migration/OfflinePlayerCurrencyMigration.java

```diff
@@ -0,0 +1,63 @@
+package dev.unnm3d.rediseconomy.currency.migration;
+
+import dev.unnm3d.rediseconomy.RedisEconomyPlugin;
+import dev.unnm3d.rediseconomy.currency.Currency;
+import dev.unnm3d.rediseconomy.currency.CurrencyMigration;
+import io.lettuce.core.ScoredValue;
+import net.milkbowl.vault.economy.Economy;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.plugin.RegisteredServiceProvider;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class OfflinePlayerCurrencyMigration extends CurrencyMigration {
+
+    private RegisteredServiceProvider<Economy> existentProvider;
```

Comment:
What's the point of using RegisteredServiceProvider?
You can directly use Vault's Economy class